decode.go

```diff
@@ -315,18 +315,35 @@ func (de *decoder) putvalue(wiretype int, val reflect.Value,
 		}
 		return de.mapEntry(val, vb)
 	case reflect.Interface:
+		data := vb[:]
+
 		// Abstract field: instantiate via dynamic constructor.
 		if val.IsNil() {
-			val.Set(de.instantiate(val.Type()))
+			id := GeneratorID{}
+			var g InterfaceGeneratorFunc
+			if len(id) < len(vb) {
```

Comment:
this appears to be the check that determines whether we should look for an interface generator. i'm interpreting it as: get the generator if the message is long enough to contain the generator ID.

what about this situation: i marshalled something that's longer than the generator ID, but i'm an attack and i'm setting the first part of my message to something like "ed.point", to force the receiver to unmarshal it using the ed25519 genrator. but the second part of my message is rubbish (an invalid ed25519 point), causing an error on the receiver's end. in fact, this may happen in a genuine use-case, some legit messages might just start with those tags by chance.

Reply:
In the case of an attacker, the decoding will fail because the point can't be constructed and then the message will be ignored (not passed through). For the legit messages, you can't indeed mix default constructors and tags because of that exact reason. Without mixing, any interface will have a tag. The worst case scenario is an error that the interface cannot be decoded meaning two parties don't have the same version or something like that.